### PR TITLE
feat: display GMT label on date publication

### PR DESCRIPTION
**Pull request by killanaca**
Added support for GMT and BST label inside date publication component (_REPLAT-1155_)

> The United Kingdom is not on Greenwich Mean Time (GMT) all year. During Daylight Saving Time (DST) the correct time zone is British Summer Time BST (offset UTC+01).

The DatePublication accepts the date and time prop as a string in ISO 8601 (UTC), parse it and  apply London time zone (according the isDateGMT prop value).

**v2 reworked**: the DatePublication component displays time zone label(based on isDateGMT) only if the user have a different time zone than the current London time zone.

Note: When we parse the date **string** with `new Date()` or `Date.parse()` methods (they are equivalent), we get the date and time in the local time.

eg: on a browser with CST timezone (China Standard Time) 

```js
new Date("2017-01-01T12:00:00.000Z") // -> Sun Jan 01 2017 20:00:00 GMT+0800 (CST)
new Date(2017,0,1,12,0) // -> Sun Jan 01 2017 12:00:00 GMT+0800 (CST)
new Date("2017-01-01T12:00:00.000Z").getHours() // -> 20
new Date("2017-01-01T12:00:00.000Z").getUTCHours() // -> 12
new Date(2017,0,1,12,0).getHours() // -> 12
new Date(2017,0,1,12,0).getUTCHours() // -> 4
```

Fixed issues with different time zone in the running env (_REPLAT-655_)

bstgmt

In case of wrong date provided to the DatePublication component, this is the behaviour:
  
  
screen shot 2018-01-10 at 17 25 25

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/2e74a1f896ba3fe6

File: packages/date-publication/date-publication.js [hunk 1]
```diff
@@ -1,21 +1,39 @@
...
+    dateJs.getUTCMonth(),
+    dateJs.getUTCDate(),
+    dateJs.getUTCHours(),
+    dateJs.getUTCMinutes()
+  );
+  let datetimeLondonTimezone = datetimeUTC;
+  if (!isGMT) {
+    datetimeLondonTimezone = addHours(datetimeUTC, 1);
+  }
+  return `${format(datetimeLondonTimezone, "dddd MMMM DD YYYY, hh:mma")} ${
```
Comment (craigbilner) [inline packages/date-publication/date-publication.js]: as agreed, we need (another) flag, to determine the "London offset" and compare this to the users offset in order to determine whether to show the suffix or not

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/6ed0a1370c982637

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/ee073bbe33281b0b

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/ca792d021d3fdf87

File: packages/article-summary/__tests__/__snapshots__/article-summary.native.test.js.snap [hunk 2]
```diff
@@ -235,7 +235,7 @@ exports[`renders an article-summary component with content including line breaks
     }
     testID="datePublication"
   >
-    Saturday July 01 2017, The Sunday Times
+    Saturday July 01 2017, 03:32pm BST, The Sunday Times
```
Comment (colincclark) [inline packages/article-summary/__tests__/__snapshots__/article-summary.native.test.js.snap]: should BST be displayed when it is BST? JIRA says not..

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/3585b5af7dfffc4e

**Comment by killanaca** [comment]
stale PR https://github.com/newsuk/times-components/pull/525

**Pull request closed without merging**